src/contracts/mixins/PermissionControllerMixin.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "../interfaces/IPermissionController.sol";
+
+abstract contract PermissionControllerMixin {
+    /// @dev Thrown when the caller is not allowed to call a function on behalf of an account.
+    error InvalidPermissions();
+
+    /// @notice Pointer to the permission controller contract.
+    IPermissionController public immutable permissionController;
+
+    constructor(
+        IPermissionController _permissionController
+    ) {
+        permissionController = _permissionController;
+    }
+
+    /// @notice Checks if the caller (msg.sender) can call on behalf of an account.
+    modifier checkCanCall(
+        address account
+    ) {
+        require(_checkCanCall(account), InvalidPermissions());
+        _;
+    }
+
+    /**
+     * @notice Checks if the caller is allowed to call a function on behalf of an account.
+     * @param account the account to check
+     * @dev `msg.sender` is the caller to check that can call the function on behalf of `account`.
+     * @dev Returns a bool, instead of reverting
+     */
+    function _checkCanCall(
+        address account
+    ) internal returns (bool) {
+        return permissionController.canCall(account, msg.sender, address(this), msg.sig);
```

Comment:
Can this return `msg.sender == account || <external call>`? That way a short-circuit will prevent an external call for what feels like a likely common case.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/efe8e2e4ad33a3a1341ba9d62fe65d5e0afcff05